### ISSUE TITLE
issues with timeline click for beginner

### ISSUE THREAD

**Issue by brendano** [AbeHandler/rookie#227]
I watched a new rookie user today:

..1. they moused over the timeline.  understood the units were months.  therefore gave the idea they could click.

..2. clicked.  a big black box appears.  they did not understand what this means.
<img width="112" alt="screen shot 2016-07-28 at 5 56 47 pm" src="https://cloud.githubusercontent.com/assets/1677/17230865/ad799c9a-54ec-11e6-8b73-138c80e9e797.png">

..3. now when in this state, trying to click elsewhere seems to cause a zero-length interval.  the UI   says "from Jun 27, 2000 — Jun 27, 2000" but that's not true, it's showing zero articles but there in fact do exist some there i think.  now they're very confused.
<img width="678" alt="screen shot 2016-07-28 at 5 57 47 pm" src="https://cloud.githubusercontent.com/assets/1677/17230907/e1b3c63e-54ec-11e6-8d4d-bdfac19bae61.png">

some ideas 

 - re 2 -- this interacts with the fact that the grabby resizer boxes are too big, or at least, they're too big in this situation.  and they overlap.  visually, what the heck does a big black box mean anyway?  if we're thinking about design tradeoffs, we certainly should aim to get right the use case of "click on the timeline and not be confused".

 - how about use a resize cursor when mouseovering the grabby box?

 - zero-length intervals should never be selected.  this seems like a bug.  minimum size should be whatever the finest granularity is ... day or month.  is there something weird with closed-open vs closed-closed intervals in the internal representation vs. user communication?  (the UI wording in the summary box sounds like a closed-closed interval.)

**Comment by brendano** [comment]
Oh another thing.  Mouseover shows month granularity.  But clicking give you a much smaller interval than month.  If the mouseover is month level, then a click ought to select exactly that month -- it's implicitly promised.  So on the assumption the granularity of resize actions (and how the interval is described in the UI) should be day-level, if a mouseover is for month X, a click should select interval [day 1 of X,  last day of X].

**Comment by brendano** [comment]
A click outside the selection should also cause that month to be selected, I think.

I remember some discussion a ways back about the difficulty of detecting a click versus a drag and whether timing comes into it.  Were you still concerned about this?  If so, fortunately I don't think it's relevant here.  If the mouseup release occurs at the same x-value as the mousedown, treat it as a "select the whole month, in the spirit of a click" event.

**Comment by AbeHandler** [comment]
ya. this is definitely something @myersnews brought up too. getting click vs. drag right was definitely a pain. i could never really get an interaction that felt right.

i think you're right that the mis-match between day and month granularity is weird. i think it makes sense to say that span T can't ever be smaller than a month. 

so an initial click opens a one month window.

re: black boxes. agree they are confusing. how about something more like...

http://www.yellowfinbi.com/Image.i4?ImageId=6354

**Comment by brendano** [comment]
> re: black boxes. agree they are confusing. how about something more like...
> http://www.yellowfinbi.com/Image.i4?ImageId=6354

yeah the three parallel vertical lines seems to be an indicator of resiziness...

**Comment by brendano** [comment]
more examples of commonly used interface elements for resizing a time interval.  we should use standard practices.

google finance ([link](https://www.google.com/finance?q=spy&ei=xVubV9ueCNLkmAH2iCs)):
<img width="325" alt="screen shot 2016-07-29 at 9 36 42 am" src="https://cloud.githubusercontent.com/assets/1677/17249913/15c8df6c-5570-11e6-86bc-aced1b10bf15.png">

Also it has
  - color changes of the below-time-line-colored-area when selected versus not.  Rookie's current method, of simply imposing gray, isn't quite as direct
  - uses only two vertical lines internally.  good, less 
  - doesn't try to constantly update during the drag -- only on mouseup release.  i wonder if this is because it's technically too hard due to too many refreshes.  i've noticed that rookie is unresponsively slow and buggy when dragging on an F-less setting.  the slowness causes bugs to surface -- for example i've seen it in a state where the mouse button is released but it is still in resize mode -- which i suspect are race conditions that were in the original code, but the slowness of this setting forces them to surface which they didn't before.

**Comment by AbeHandler** [comment]
Easy (and fixed):

 - update on mouse up only (done)
 - thinner black bars on side, lines on top of rectangle
 - zero length selection should be impossible now. initial click = 1 month
 - resize cursor on handles
 - month level granularity. no more day granularity

Harder (lower priority, but certainly doable): 

 - color changes of the below-time-line-colored-area when selected versus not. Rookie's current method, of simply imposing gray, isn't quite as direct
